source/_integrations/roku.markdown

```diff
@@ -130,6 +127,44 @@ action:
       media_content_type: channel
 ```
 
+## Content Deeplinking
+
+The `media_player.play_media` service may be used to deep link to content within an application.
+
+| Service data attribute | Optional | Description | Example |
+| ---------------------- | -------- | ----------- | ------- |
+| `entity_id` | no | Target a specific media player. To target all media players, use `all`. |                                                                                                                    |
+| `media_content_id` | no | A media identifier. | 291097
+| `media_content_type` | no | A media type. | channel
+| `extra.content_id` | no | A unique content identifier passed to app. | 8e06a8b7-d667-4e31-939d-f40a6dd78a88
+| `extra.media_type` | no | A media type passed to app. Should be one of `app`, `movie`, `episode`, `season`, `series`, `shortFormVideo`, `special`, `live` | movie
+
+### Example
+
+```yaml
+action:
+  - service: media_player.play_media
+    target:
+      entity_id: media_player.roku
+    data:
+      media_content_id: 291097
+      media_content_type: app
+      extra:
+        content_id: 8e06a8b7-d667-4e31-939d-f40a6dd78a88
+        media_type: movie
+```
+
+### Obtaining Content IDs
+
+Content IDs are unique to each streaming service and vary in format but are often part of the video webpage URL. Here are some examples:
+
+| Service | App ID | URL Format | Content ID | Media Type
+| ------- | ------ | ---------- | ---------- | ---------- |
+| Disney Plus | 291097 | disneyplus.com/video/8e06a8b7-d667-4e31-939d-f40a6dd78a88 | 8e06a8b7-d667-4e31-939d-f40a6dd78a88 | movie
+| Hulu | 2285 | hulu.com/series/american-dad-977c8e25-cde0-41b7-80ce-e746f2d2093f | american-dad-977c8e25-cde0-41b7-80ce-e746f2d2093f | series
+| Spotify | 22297 | N/A | spotify:playlist:abcdefghij0123456789XY | app
```

Comment:
The Spotify URL format is this: https://open.spotify.com/playlist/5xddIVAtLrZKtt4YGLM1SQ, and the `mediaType` is not app, this was a misunderstanding on the Reddit thread, it would be playlist in this case, and if is a song would be `track`

Reply:
In fact the `mediaType` parameter should be optional, most apps will not bother if you don't send it, including YouTube and Spotify

Reply:
update spotify example.  I had based the mediaType being required on the docs. https://developer.roku.com/docs/developer-program/discovery/implementing-deep-linking.md#using-ecp-commands-for-testing-deep-linking

Reply:
I think is good to have, but in reality, most of the apps do not look for it